### ISSUE TITLE
As a self-hosted user, I would like to be able to access my own Hipchat instance's web client

### ISSUE THREAD

**Issue by schredder** [catdad/unofficial-hipchat#8]
This can usually be accessed from `https://${MY_HIPCHAT_HOST}/chat`

**Comment by catdad** [comment]
@schredder I looked at how the official apps do this. They basically have box where you can type in your own URL, and then that it always attempt to log in to that URL instead of the official one.

Would you be willing to run a test build to make sure that it works? I don't have access to a self hosted server to test it out.

**Comment by schredder** [comment]
Yeah, I'd be willing to test it out for you.

**Comment by catdad** [comment]
@schredder The two commits above implement a working version, but it is pretty ugly right now. You can create a build now if you are in a rush, but if not, I will provide a build when it's a bit more polished. I won't have time to finish it up until later tomorrow or this weekend.

**Comment by schredder** [comment]
:+1: I'm not in a rush, but I'll play with it. Thanks!

**Comment by catdad** [comment]
I'm pretty happy with this now. I might fidget with it a bit more, but if all works, I will try to release it officially by Monday.

https://drive.google.com/open?id=0BzobV6zzoAikczMwOF92S05lT2s

You may need to uninstall the actual app in order to test this. Unzip the file and load it in Chrome as an unpacked app.

There is now a hamburger button in the top right, next to the minimize button, to open up the app settings. You can enter a new URL to use there.

**Comment by schredder** [comment]
Thanks much. I built it from the repo on Friday to try it out. I'm going to try to throw something together for this: The self-hosted and SaaS Hipchats' login pages are identical, so it isn't terribly obvious what you're logging in to.

**Comment by catdad** [comment]
Yeah, I thought about that, but then I referred to how the official apps work. In those, you change a URL in the settings, but everything stays the same... you just kind of trust it that it went to the correct URL. If you stay on the web link, then your login just wouldn't work.

Off the top of my head, I would think sending a notification saying "You are logging into HipChat Server at ..." (or something like that) would be simple and not much effort. Although that could be annoying.

**Comment by catdad** [comment]
Published in v1.0.15336.218.

**Comment by schredder** [comment]
```
________$$$$
_______$$__$
_______$___$$
_______$___$$
_______$$___$$
________$____$$
________$$____$$$
_________$$_____$$
_________$$______$$
__________$_______$$
____$$$$$$$________$$
__$$$_______________$$$$$$
_$$____$$$$____________$$$
_$___$$$__$$$____________$$
_$$________$$$____________$
__$$____$$$$$$____________$
__$$$$$$$____$$___________$
__$$_______$$$$___________$
___$$$$$$$$$__$$_________$$
____$________$$$$_____$$$$
____$$____$$$$$$____$$$$$$
_____$$$$$$____$$__$$
_______$_____$$$_$$$
________$$$$$$$$$$
```
